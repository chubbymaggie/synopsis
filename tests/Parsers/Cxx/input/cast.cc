typedef int Foo;

void func()
{
  (Foo)1;
  (Foo*)1;
  (const Foo&)1;
}
