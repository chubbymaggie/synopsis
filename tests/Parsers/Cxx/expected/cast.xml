<?xml version='1.0' encoding='ISO-8859-1'?>
<ast>
 <declarations>
  <instance accessibility="0" class="Synopsis.AST.Typedef" constr="0" file="Parsers/Cxx/input/cast.cc" language="C++" line="1" name="Foo" type="typedef">
   <alias>
    <instance class="Synopsis.Type.Base" language="C++" name="int"/>
   </alias>
  </instance>
  <instance accessibility="0" class="Synopsis.AST.Function" file="Parsers/Cxx/input/cast.cc" language="C++" line="3" name="func()" realname="func" type="function">
   <returnType>
    <instance class="Synopsis.Type.Base" language="C++" name="void"/>
   </returnType>
  </instance>
 </declarations>
 <types>
  <instance class="Synopsis.Type.Declared" language="C++" name="Foo">
   <declaration/>
  </instance>
  <instance class="Synopsis.Type.Declared" language="C++" name="func()">
   <declaration/>
  </instance>
 </types>
 <files>
  <instance class="Synopsis.AST.SourceFile" filename="Parsers/Cxx/input/cast.cc" is_main="1" language="C++">
   <declarations/>
   <macro_calls/>
  </instance>
 </files>
</ast>
